array.c

```diff
@@ -1016,14 +1016,14 @@ rb_to_array(VALUE ary)
  *  call-seq:
  *    Array.try_convert(object) -> object, new_array, or nil
  *
- *  If +object+ is an +Array+ object, returns +object+.
+ *  Attempts to convert the given +object+ to an +Array+ object:
  *
- *  Otherwise if +object+ responds to <tt>:to_ary</tt>,
- *  calls <tt>object.to_ary</tt> and returns the result.
+ *  - If +object+ is an +Array+ object, returns +object+.
+ *  - Otherwise if +object+ responds to <tt>:to_ary</tt>,
+ *    returns <tt>object.to_ary</tt>.
+ *  - Otherwise returns +nil+.
  *
- *  Returns +nil+ if +object+ does not respond to <tt>:to_ary</tt>
- *
- *  Raises an exception unless <tt>object.to_ary</tt> returns an +Array+ object.
+ *  Raises TypeError unless <tt>object.to_ary</tt> returns an +Array+ object or +nil+.
```

Comment:
I think we should move this under the second bullet point since `object` could be an array or nil, then the return value of `object.to_ary` is never checked.

Reply:
Modified.